File: aac_library/src/main/java/com/example/aac_library/base/WanBaseResponse.java
package com.example.aac_library.base;

import com.google.gson.annotations.SerializedName;

public class WanBaseResponse<T>{

    @SerializedName("errorCode")
    private int code;

    @SerializedName("errorMsg")
    private String msg;

    @SerializedName("data")
    private T data;


    public int getCode() {
        return code;
    }

    public String getMsg() {
        return msg;
    }

    public T getData() {
        return data;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public void setData(T data) {
        this.data = data;
    }
}

File: aac_library/src/main/java/com/example/aac_library/base/WanBaseResponse.kt
package com.example.aac_library.base

import com.google.gson.annotations.SerializedName

open class WanBaseResponse<T>(
    @SerializedName("errorCode") var code: Int = 0,

    @SerializedName("errorMsg") var msg: String? = null,

    @SerializedName("data") var data: T? = null
)
